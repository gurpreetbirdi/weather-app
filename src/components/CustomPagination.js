import React from 'react';
import PropTypes from 'prop-types';
import { Grid, IconButton } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import clsx from 'clsx';

const useStyles = makeStyles(() => ({
  icon: { justifyContent: 'space-between' },
  hidden: { visibility: 'hidden' },
}));

const CustomPagination = ({ gotoPage, pagination }) => {
  const classes = useStyles();
  const { hasPreviousPage, hasNextPage } = pagination;

  return (
    <Grid item container className={classes.icon}>
      <IconButton
        id="back-btn"
        size="medium"
        className={clsx(!hasPreviousPage && classes.hidden)}
        onClick={() => gotoPage('back')}>
        <ArrowBackIcon fontSize="inherit" />
      </IconButton>
      <IconButton
        id="next-btn"
        size="medium"
        className={clsx(!hasNextPage && classes.hidden)}
        onClick={() => gotoPage('next')}>
        <ArrowForwardIcon fontSize="inherit" />
      </IconButton>
    </Grid>
  );
};

CustomPagination.propTypes = {
  gotoPage: PropTypes.func,
  pagination: PropTypes.shape({
    hasPreviousPage: PropTypes.bool,
    hasNextPage: PropTypes.bool,
  }),
};

export default CustomPagination;
