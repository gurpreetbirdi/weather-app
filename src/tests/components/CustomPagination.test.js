import React from 'react';
import { shallow } from 'enzyme';

import CustomPagination from '../../components/CustomPagination';

const pagination = {
  hasPreviousPage: true,
  hasNextPage: true,
};

describe('<CustomPagination />', () => {
  const gotoPage = jest.fn();
  const wrapper = shallow(<CustomPagination pagination={pagination} gotoPage={gotoPage} />);

  test('should render CustomPagination correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call gotoPage with next on click', () => {
    wrapper.find('#next-btn').simulate('click');
    expect(gotoPage).toHaveBeenCalledWith('next');
  });

  test('should call gotoPage with back on click', () => {
    wrapper.find('#back-btn').simulate('click');
    expect(gotoPage).toHaveBeenCalledWith('back');
  });
});
