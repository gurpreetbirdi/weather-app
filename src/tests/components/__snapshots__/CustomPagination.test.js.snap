// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomPagination /> should render CustomPagination correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-icon-1"
  container={true}
  item={true}
>
  <WithStyles(ForwardRef(IconButton))
    className=""
    id="back-btn"
    onClick={[Function]}
    size="medium"
  >
    <ArrowBackIcon
      fontSize="inherit"
    />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(IconButton))
    className=""
    id="next-btn"
    onClick={[Function]}
    size="medium"
  >
    <ArrowForwardIcon
      fontSize="inherit"
    />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(Grid))>
`;
